0.99.0: kind of ABI/API changes issue

Looks like ABI/API changes in 0.99.0 are quitre subsptantial and by this libgit2-glib can ot be build 0.99.0. SONAME of the library changed as well (which is good in this case).

However pkgconfig file still is libgit2.pc. I think that probably it would be good to change it to libgit2-1.pc to not confuse pkgconfig detection.

> `./libgit2-glib/ggit-patch.c:197:5: error: token ""0.99"" is not valid in preprocessor expressions`

This seems like a poor assumption in caller code that this would always be numeric.

> ../libgit2-glib/ggit-cred-plaintext.c:182:40: error: invalid use of incomplete typedef 'git_cred' {aka 'git_credential'}

This seems like a problem.

> Especially with v1.0 I could see us moving to libgit2-1.0.pc.

Sure, this seems reasonable.  Obviously it's too late to put the genie back in the 0.99 bottle though.

> ../libgit2-glib/ggit-config.c:303:8: error: too few arguments to function 'git_config_add_file_ondisk'
>  303 |  ret = git_config_add_file_ondisk (_ggit_native_get (config),

This one is confusing, though.  Blame shows that this function's signature hasn't changed in years.

> > ../libgit2-glib/ggit-cred-plaintext.c:182:40: error: invalid use of incomplete typedef 'git_cred' {aka 'git_credential'}

> This seems like a problem.

Aha.  There was a breaking change here in #5223.  I'm going to move those things into `include/git2/credential.h`

--- a/include/git2/deprecated.h
+++ b/include/git2/deprecated.h
@@ -41,6 +41,13 @@
  */
 #ifndef GIT_DEPRECATE_HARD
 
+/*
+ * The credential structures are now opaque by default, and their
+ * definition has moved into the `sys/credential.h` header; include
+ * them here for backward compatibility.
+ */
+#include "sys/credential.h"
+
 /**
  * @file git2/deprecated.h
  * @brief libgit2 deprecated functions and values

